### PR TITLE
Update repo to work with Terraform 0.13.x

### PR DESCRIPTION
**Pull request by robmorgan**
1. Update CircleCI to use Terraform 0.13.x.
1. Upgraded the Google Providers to 3.43.0.
1. Dropped GAE configuration options.
1. Update the version constraint on all examples to require at least 0.12.26.
1. Update pre-commit hooks to add `gofmt`.
1. Upgrade the tests to support Terratest v0.30.23

In general, TF 0.13.x seems backwards compatible from a testing perspective: I did not need to change anything in `modules` in this PR.

### CODE REVIEW

**Comment by robmorgan** [comment]
Builds are currently failing.

**Comment by robmorgan** [comment]
Tests passed in https://app.circleci.com/pipelines/github/gruntwork-io/terraform-google-sql/93/workflows/3c5acbbd-765b-4a27-b9c4-3057e920c6eb/jobs/191

File: modules/cloud-sql/main.tf [hunk 1]
```diff
@@ -39,10 +46,9 @@ resource "google_sql_database_instance" "master" {
   database_version = var.engine
 
   settings {
-    tier                        = var.machine_type
-    activation_policy           = var.activation_policy
-    authorized_gae_applications = var.authorized_gae_applications
-    disk_autoresize             = var.disk_autoresize
+    tier              = var.machine_type
+    activation_policy = var.activation_policy
+    disk_autoresize   = var.disk_autoresize
```
Comment (brikis98) [inline modules/cloud-sql/main.tf:51]: Are these Terraform 0.13 changes or new GCP provider changes? I'm guessing the latter?

File: modules/cloud-sql/variables.tf [hunk 2]
```diff
@@ -69,12 +69,6 @@ variable "authorized_networks" {
   # ]
 }
 
-variable "authorized_gae_applications" {
-  description = "A list of Google App Engine (GAE) project names that are allowed to access this instance."
-  type        = list(string)
-  default     = []
-}
-
```
Comment (brikis98) [inline modules/cloud-sql/variables.tf:77]: Make sure the release notes capture these changes and have a migration guide!

**Review by brikis98 — APPROVED** [review]
LGTM!

File: modules/cloud-sql/main.tf (hunk 1 above)
Comment (robmorgan) [reply modules/cloud-sql/main.tf:51]: Product management changes actually! I was going to have to work around a bit of the Google App Engine stuff, so I'm actually planning to deprecate it! I assuming its not the direction most of our customers are going in and none of the Google maintained Open Source modules offer support for it either.

File: modules/cloud-sql/variables.tf (hunk 2 above)
Comment (robmorgan) [reply modules/cloud-sql/variables.tf:77]: Yep don't worry I have a guide prepared!

**Comment by robmorgan** [comment]
@brikis98 thanks for the review. I'll release next week!

**Pull request merged**